releasenotes/notes/conservative-commutation-checking-b728e7b6e1645615.yaml

```
@@ -1,18 +1,19 @@
 ---
 fixes:
   - |
-    Commutation relations of :class:`~.circuit.Instruction`\ s with float-only ``params``
-    were eagerly cached by the :class:`.CommutationChecker`, using the ``params`` as key to
-    query the relation. This could lead to faulty results, if the instruction's definition
-    depended on additional information that just the :attr:`~.circuit.Instruction.params`
-    attribute, such as e.g. the case for :class:`.PauliEvolutionGate`.
-    This behavior is now fixed, and the commutation checker only conservatively caches
-    commutations for Qiskit-native standard gates. This can incur a performance cost if you were
-    relying on your custom gates being cached, however, we cannot guarantee safe caching for
-    custom gates, as they might rely on information beyond :attr:`~.circuit.Instruction.params`.
+    Previously, the :class:`.CommutationChecker` eagerly cached the commutation relations of
+    :class:`~.circuit.Instruction`\ s with  float-only ``params`` as key to query the relation.
+    This could lead to incorrect results if the instruction's definition
+    depended on additional information beyond just the :attr:`~.circuit.Instruction.params`
+    attribute, as in the case of :class:`.PauliEvolutionGate`.
+    This behavior is now fixed, and the commutation checker conservatively caches
```

Actually, this should be "eagerly" -- we do not cache them for _non-Qiskit_ gates, but for the standard gates, where we know all info is in `params` we still eagerly cache.
```suggestion
    This behavior is now fixed, and the commutation checker eagerly caches
```

---

Agreed, "eagerly" is better (or, to be even more precise, the commutation checker is now more conservative with respect to _which_ gates it hashes, but is still eager in _how_ it hashes them, but I believe that this is now clear from the context and does not need to be mentioned explicitly).